Dynavity/Dynavity/view/CanvasElementMapView.swift

```diff
@@ -9,31 +9,54 @@ import SwiftUI
 
 struct CanvasElementMapView: View {
     @ObservedObject var viewModel: CanvasViewModel
-    @Binding var scaleFactor: CGFloat
+
+    private var dragGesture: some Gesture {
+        DragGesture()
+            .onChanged { value in
+                viewModel.moveSelectedCanvasElement(by: value.translation)
+            }
+    }
+
+    private func transformToView(element: CanvasElementProtocol) -> some View {
+        Group {
+            switch element {
+            case let imageCanvasElement as ImageCanvasElement:
+                ImageCanvasElementView(imageCanvasElement: imageCanvasElement)
+            case let pdfCanvasElement as PDFCanvasElement:
+                PDFCanvasElementView(pdfCanvasElement: pdfCanvasElement)
+            case let textBlock as TextBlock:
+                TextBlockView(textBlock: textBlock)
+            case let markupTextBlock as MarkupTextBlock:
+                MarkupTextBlockView(markupTextBlock: markupTextBlock)
+            default:
+                CanvasElementView(element: element)
+            }
+        }
+    }
+
+    private func isSelected(_ element: CanvasElementProtocol) -> Bool {
+        viewModel.selectedCanvasElementId == element.id
+    }
 
     var body: some View {
         ZStack {
             ForEach(viewModel.canvas.canvasElements, id: \.id) { element in
-                // The group allows us to have common view modifiers.
-                Group {
-                    switch element {
-                    case let imageCanvasElement as ImageCanvasElement:
-                        ImageCanvasElementView(imageCanvasElement: imageCanvasElement)
-                    case let pdfCanvasElement as PDFCanvasElement:
-                        PDFCanvasElementView(pdfCanvasElement: pdfCanvasElement)
-                    case let textBlock as TextBlock:
-                        TextBlockView(textBlock: textBlock)
-                    case let markupTextBlock as MarkupTextBlock:
-                        MarkupTextBlockView(markupTextBlock: markupTextBlock)
-                    default:
-                        CanvasElementView(element: element)
+                ZStack {
+                    transformToView(element: element)
+                        .frame(width: element.width, height: element.height)
+                        .addCardOverlay()
+                        .onTapGesture {
+                            viewModel.select(canvasElement: element)
+                        }
+                        .gesture(isSelected(element) ? dragGesture : nil)
+                    if isSelected(element) {
+                        SelectionOverlayView(element: element, viewModel: viewModel)
```

Comment:
Can this be added using `.overlay(...)` or does that not work?

Reply:
Seems to be functionally the same. We'll go with this since it's more expressive of our intent in this specific situation, thanks!